lib/streamlit/server/server.py

```diff
@@ -400,9 +402,13 @@ def _loop_coroutine(self, on_started=None):
             while not self._must_stop.is_set():
 
                 if self._state == State.WAITING_FOR_FIRST_BROWSER:
-                    pass
+                    yield tornado.gen.WaitIterator(
```

Comment:
Could you add a brief comment explaining what this is doing?

Reply:
I, having only briefly skimmed the tornado docs, didn't know the semantic meaning of `tornado.gen.maybe_future(self._must_stop.wait()),`, nor that `next()` is the first of those two futures to complete, rather than some sort of toggling between them, for example.
A comment along the lines of "Waits for one of the _must_stop and _has_connection events to occur" would be sufficient. Just enough context to understand the basic control flow, for me and my teammates who aren't familiar with tornado to not have to look it up every time.

Reply:
`maybe_future` function calls needed because of https://github.com/tornadoweb/tornado/issues/3001, it is a workaround, it does not do anything.

So as I understand the `WaitIterator` is not a clear class name, It doesn't explain its own purpose.

I can suggest that It can be used native `asyncio.wait` function, where need to add explicitly  the `FIRST_COMPLETED` constant, and does not need to wrap coroutines/futures (and probably needs no explanation in comment). 

basically:
```
yield tornado.gen.convert_yielded(asyncio.wait([
    self._must_stop.wait(), 
    self._has_connection.wait()
], return_when=asyncio.FIRST_COMPLETED))
```

This modification makes the code obvious?

Reply:
Yes, I think that would be sufficient